src/Hazelcast.Net/Clustering/MemberConnection.cs

```diff
@@ -142,8 +145,14 @@ public Action<ClientMessage> ReceivedEvent
 
         /// <summary>
         /// Gets the network address the client is connected to.
+        /// <para>In case of unisocket mode, gets the <see cref="RemoteAddress"/></para>
+        /// </summary>
+        public NetworkAddress Address { get { return _networkingOptions.SmartRouting || _networkingOptions.Cloud.Enabled ? _address : RemoteAddress; } }
+
+        /// <summary>
+        /// Gets network address responed by the member
```

Comment:
`// Gets the network address returned by the member.`

And, it's not "returned" by the member... or is it? Or is it... "provided by the members list"?